init.lua

```diff
@@ -1,6 +1,17 @@
 castle_masonry = {}
 
+castle_masonry.sounds = {}
```

Comment:
It's better to initialize this to `nil` and raise an error below when neither default nor mcl were found because the functions that are called from within do not exist. e.g. `castle_masonry.sounds.node_sound_stone_defaults(),`.

Aside from that the code looks good. Thank you for the changes.

Reply:
Maybe another game without `default` or without `mcl_sounds` wants to use this mod, I think we should not force the error, I think setting `castle_masonry.sounds = {}` avoids this error and allows more compatibility

Reply:
how does it avoid the error? You're trying to call a nil value when neither default not mtg are present.
`castle_masonry.sounds.node_sound_stone_defaults()` is used and not
`castle_masonry.sounds.node_sound_stone_defaults` which is a crucial difference.

Reply:
Oh I see what you mean. Mods would then overwrite `castle_masonry.sounds` manually and call the registration functions afterwards. That makes sense but is not very reliable either because the recipe items are not guaranteed.

Reply:
I suppose that if another mod (without `default` and without `mcl_sounds`) tries to use `castle_masonry` it will also add its materials and therefore its guaranteed recipes to the global table `castle_masonry.materials = {}` using `table.insert(castle_masonry.materials, def)`